### PR TITLE
Support malformed numbers in synthetic _source

### PR DESCRIPTION
**Pull request by nik9000**
This adds support for `ignore_malformed` to numeric fields other than `scaled_float` in synthetic `_source`. Their values are saved to a stored field and loaded to render the `_source`.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/index/mapper/NumberFieldMapperTests.java [hunk 1]
```diff
@@ -379,6 +383,7 @@ protected final class NumberSyntheticSourceSupport implements SyntheticSourceSup
         private final Function<Number, Number> round;
         private final Long nullValue = usually() ? null : randomNumber().longValue();
         private final boolean coerce = rarely();
+        private final boolean ignoreMalformed = rarely();
```
Comment (romseygeek) [inline server/src/test/java/org/elasticsearch/index/mapper/NumberFieldMapperTests.java:386]: Do we have a specific test for this somewhere?

Comment (nik9000) [reply server/src/test/java/org/elasticsearch/index/mapper/NumberFieldMapperTests.java:386]: It's run `rarely` like null value support or coerce or whatever. I could could run it on each run in a new test if you'd like. That way we're more likely to catch regressions faster....

Comment (romseygeek) [reply server/src/test/java/org/elasticsearch/index/mapper/NumberFieldMapperTests.java:386]: Yeah, I think I'd be happier with explicit tests for each configuration.

Comment (nik9000) [reply server/src/test/java/org/elasticsearch/index/mapper/NumberFieldMapperTests.java:386]: I've opened #90565 so we'll have known examples to test in addition to some tests each iteration.

**Review by nik9000 — COMMENTED** [review]
@romseygeek could you have another look at this one now that I've merged in my `ignore_malformed` testing?

File: modules/mapper-extras/src/test/java/org/elasticsearch/index/mapper/extras/ScaledFloatFieldMapperTests.java [hunk 2]
```diff
@@ -337,75 +337,76 @@ protected Object generateRandomInputValue(MappedFieldType ft) {
...
-                    return new SyntheticSourceExample(v.v1(), v.v2(), this::mapping);
-                }
-                List<Tuple<Double, Double>> values = randomList(1, maxValues, this::generateValue);
-                List<Double> in = values.stream().map(Tuple::v1).toList();
-                List<Double> outList = values.stream().map(Tuple::v2).sorted().toList();
-                Object out = outList.size() == 1 ? outList.get(0) : outList;
-                return new SyntheticSourceExample(in, out, this::mapping);
-            }
+    protected SyntheticSourceSupport syntheticSourceSupport(boolean ignoreMalformed) {
+        assumeFalse("scaled_float doesn't support ignore_above with synthetic _source", ignoreMalformed);
```
Comment (romseygeek) [inline modules/mapper-extras/src/test/java/org/elasticsearch/index/mapper/extras/ScaledFloatFieldMapperTests.java:341]: `ignore_malformed` not `ignore_above`, right?

**Review by romseygeek — APPROVED** [review]
LGTM, thanks for extending the test coverage.

**Pull request merged**